---
layout: post
title: "[파이썬] 예외 처리와 예외 감지"
description: " "
date: 2023-09-01
tags: [python]
comments: true
share: true
---

예외 처리(Exception Handling)는 프로그램 실행 도중 발생할 수 있는 예상치 못한 상황에 대비하여 적절한 조치를 취하는 프로그래밍 기법입니다. Python은 강력하고 유연한 예외 처리 기능을 제공하여 프로그래머가 예외 상황에 대응할 수 있도록 도와줍니다. 

## 예외 처리의 필요성

프로그램을 작성하다보면 다양한 예외 상황을 다루어야 할 때가 있습니다. 예를 들어, 사용자 입력에 대한 유효성 검사를 수행하는 프로그램을 작성한다고 가정해 봅시다. 사용자가 잘못된 형식의 입력을 하거나 필수 입력 항목을 빠뜨린 경우, 프로그램은 해당 상황을 처리하고 사용자에게 알려주어야 합니다. 이러한 상황에서 예외 처리는 매우 유용한 도구입니다.

## 예외 처리 방법

Python에서 예외 처리는 `try-except` 문을 사용하여 구현할 수 있습니다. 예외가 발생할 가능성이 있는 코드는 `try` 블록 안에 작성되고, 예외가 발생하면 해당 예외 유형에 대응하는 `except` 블록이 실행됩니다. 

다음은 파일을 열고 읽어오는 예제 코드입니다. 파일이 존재하지 않거나 열 수 없는 경우를 처리하는 예외 처리 코드를 포함합니다.

```python
try:
    file = open("myfile.txt", "r")
    content = file.read()
    print(content)
except FileNotFoundError:
    print("파일을 찾을 수 없습니다.")
except IOError:
    print("파일을 열 수 없습니다.")
finally:
    file.close()

```

위 코드에서 `try` 블록에서 파일을 열고 읽어오는 과정을 시도합니다. 만약 파일이 존재하지 않거나 열 수 없는 경우, 해당하는 예외 유형의 `except` 블록이 실행됩니다. 마지막으로 `finally` 블록은 예외 발생 여부와 상관없이 항상 실행되는 코드 블록입니다. 파일을 열었다면, 최종적으로 파일을 닫기 위해 `finally` 블록 내에서 `file.close()`를 호출합니다.

## 예외 감지(Detecting Exceptions)

예외를 감지하기 위해서는 `try-except` 구문을 사용하는데, 예외 객체를 통해 발생한 예외에 대한 정보를 얻을 수 있습니다. 주로 예외 객체의 `__str__()` 메서드를 활용하여 예외에 대한 정보를 출력합니다.

다음은 0으로 나누는 경우 `ZeroDivisionError`가 발생할 수 있는 예제 코드입니다.

```python
try:
    result = 10 / 0
except ZeroDivisionError as e:
    print("에러 발생:", e)

```

위 코드는 0으로 나눌 때 발생하는 `ZeroDivisionError`를 처리합니다. `except` 블록에서 예외 객체를 `e`로 받아와 에러 메세지를 출력합니다.

## 예외 처리의 중요성

적절한 예외 처리는 프로그램의 안정성을 향상시키고 예상치 못한 예외 상황에 대응할 수 있도록 도와줍니다. 예외 처리를 통해 프로그램이 비정상적으로 종료되지 않고 예외 상황을 정상적으로 처리할 수 있습니다. 또한, 예외에 대한 필요한 조치를 취함으로써 사용자에게 친절한 애플리케이션을 제공할 수 있습니다.

올바른 예외 처리를 통해 프로그램의 견고성과 안정성을 향상시키세요.

이상으로 Python에서의 예외 처리와 예외 감지에 대한 내용을 알아보았습니다. 예외 처리 기능을 적절히 활용하여 프로그램을 작성하고 예외 상황을 철저히 다룰 수 있도록 노력해보세요!